### ISSUE TITLE
User's Guide 26.3 OMShell Examples section proposed changes

### ISSUE THREAD

**Issue by mjschuh** [OpenModelica/OpenModelica#11021]
### Description

The User's Guide 26.3 OMShell Examples section is very helpful and could be enhanced to be even more helpful.  While trying to learn how to use the OMShell by reading and rereading the OpenModelica User's Guide several times, I had a lot of trouble figuring things out from reading the text.  Here are a few things that I think would be helpful for others trying to learn how to use the OMShell.

### 26.3 Examples

Suggestions:
<ul>
<li>At the end of the first paragraph that ends in "First we start the system ... or command window of some of the other tools." add "The examples below were run using the <code>OMShell-terminal</code> command."  I kept thinking that things were not working for me because I was running different versions of OMshell that did not have the same prompt.  It would have helped me figure out what I was doing wrong if I was certain that I was running the same OMShell that was used for the Section 26.3 Examples.</li>

<li>Change "We type in a very small model:" to "We type in a very small model and save in a file named <code>Test.mo</code>:"</li>

<li>I found that the <code>instantiateModel(Test)</code> step could not find the <code>Test.mo</code> file.  Before the <code>instantiateModel(Test)</code> step, I suggest adding "<code>loadFile("Test.mo")</code>."</li>

<li>The <code>plotall()</code> step did not work for me.  I finally figured out that the <b>A</b> in "all" needs to be capitalized.  Using <code>plotAll()</code> worked for me.</li>

</ul>

### 26.1 OpenModelica Scripting Commands

I think it would be very helpful if this section included descriptions of each of the commands.  It would also be helpful if there were examples of the commands.  It appears that not all of the <code>input</code> arguments are required.  It would be helpful if this sections started with a description of when an <code>input</code> parameter is required or not.

**Comment by mjschuh** [comment]
Here is a link to the [pdf version of the OpenModelica User Guide](https://openmodelica.org/doc/OpenModelicaUsersGuide/OpenModelicaUsersGuide-1.21.pdf).  Section 26.3 Examples begins on the document page number 428 (pdf page 437) in OpenModelica User’s Guide, Release v1.21.0-2-g0bc3d29bd9.  Here is a link to the [Examples section in the HTML version of the User Guide](https://openmodelica.org/doc/OpenModelicaUsersGuide/1.21/scripting_api.html#examples).

**Comment by adeas31** [comment]
> - At the end of the first paragraph that ends in "First we start the system ... or command window of some of the other tools." add "The examples below were run using the OMShell-terminal command." I kept thinking that things were not working for me because I was running different versions of OMshell that did not have the same prompt. It would have helped me figure out what I was doing wrong if I was certain that I was running the same OMShell that was used for the Section 26.3 Examples.

The statement is correct. It doesn't matter if you use OMShell-terminal, OMNotebook or any other OM tool. The examples should run in all of them. I updated the statement to `Run the examples below using OMShell or OMNotebook.`

> - Change "We type in a very small model:" to "We type in a very small model and save in a file named Test.mo:"

You type the model in the command window and there is no need to save it to the file. I made it more clear by changing it to `We type in a very small model in the command window:`

> - I found that the instantiateModel(Test) step could not find the Test.mo file. Before the instantiateModel(Test) step, I suggest adding "loadFile("Test.mo")."

Not needed.

> - The plotall() step did not work for me. I finally figured out that the A in "all" needs to be capitalized. Using plotAll() worked for me.

Fixed it.

> I think it would be very helpful if this section included descriptions of each of the commands. It would also be helpful if there were examples of the commands. It appears that not all of the input arguments are required. It would be helpful if this sections started with a description of when an input parameter is required or not.

It requires some more effort to add examples for these commands. There is another documentation for these commands https://build.openmodelica.org/Documentation/OpenModelica.Scripting.html. I linked this from the users guide.

All the input parameters that have a default value are not required.

**Comment by mjschuh** [comment]
Thank you for fixing <code>plotAll()</code> in the documentation.

Here is what I see when I try to use <code>instantiateModel(Test)</code> in the Windows version of OMShell

```
OMShell 1.1 Copyright Open Source Modelica Consortium (OSMC) 2002-2022
Distributed under OMSC-PL and GPL, see www.openmodelica.org

Connected to OpenModelica v1.20.0-dev-345-g093ed4cfda (64-bit)
To get help on using OMShell and OpenModelica, type "help()" and press enter.

>> instantiateModel(Test)
""

>> class Test "Testing"
Error occurred building AST
Syntax Error
[<interactive>:1:0-1:0:writable] Error: No viable alternative near token: <EOF>

>> 
```
The Ubuntu version of <code>OMShell-terminal</code> has a slightly different behavior
```
mschuh@ARDWG0121101002:~/Training/omshell$ OMShell-terminal
OMShell Copyright 1997-2023, Open Source Modelica Consortium (OSMC)
Distributed under OMSC-PL and GPL, see www.openmodelica.org

To get help on using OMShell and OpenModelica, type "help()" and press enter
Set shortOutput flag: true

>>> instantiateModel(Test)
""
"Error: Failed to load package Test (default) using MODELICAPATH /home/mschuh/.openmodelica/libraries/.
Error: Class Test not found in scope <top>.
"
>>>
```

Using <code>instantiateModel(Test)</code> does not appear to work for me.  If I put a copy of `Test.mo` in `~/.openmodelica/libraries`, using <code>instantiateModel(Test)</code> will load it.  Is there a different command from <code>instantiateModel(Test)</code> that I should be using?  Am I doing something wrong?  

> All the input parameters that have a default value are not required.

It would be helpful if this detail was included in the documentation in Section 26.1

Thanks,
  Michael

**Comment by adeas31** [comment]
You first need to type/load the model in the command window/OMShell.
Type/paste the model directly in OMShell and press enter.
